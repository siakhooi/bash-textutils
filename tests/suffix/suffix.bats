setup(){
    load '../common-setup'
    common_setup

    testDatafile1=$BATS_TEST_DIRNAME/test-data-1
    testDatafile2=$BATS_TEST_DIRNAME/test-data-2

    outputExpected=$BATS_TEST_FILENAME.$BATS_TEST_DESCRIPTION.expected
    outputActual="$TEST_TEMP_DIR/$(basename "$BATS_TEST_FILENAME").$BATS_TEST_DESCRIPTION.actual"
}
@test "suffix-0-no-argument" {
    run bash -c "suffix | tee $outputActual"
    assert_output - <$outputExpected
}
@test "suffix-1" {
    run bash -c "cat $testDatafile1 | suffix ' <<<<' | tee $outputActual"
    assert_output - <$outputExpected
}
@test "suffix-2" {
    run bash -c "suffix ' <<<<' $testDatafile1 | tee $outputActual"
    assert_output - <$outputExpected
}
@test "suffix-3-invalid-arguments" {
    run bash -c "suffix 'a' 'b' 'c' | tee $outputActual"
    assert_output - <$outputExpected
}
@test "suffix-4-non-exists-file" {
    run bash -c "suffix ' <<<<' ./non-exists-file | tee $outputActual"
    assert_output - <$outputExpected
}
@test "suffix-5-directory" {
    run bash -c "suffix ' <<<<' /etc | tee $outputActual"
    assert_output - <$outputExpected
}
@test "suffix-6" {
    run bash -c "suffix ' <<<<' $testDatafile2 | tee $outputActual"
    assert_output - <$outputExpected
}
