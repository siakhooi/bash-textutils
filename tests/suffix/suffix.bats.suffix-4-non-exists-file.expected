Input file ./non-exists-file is not readable.
